js/modules/k6/experimental/tracing/client.go

```diff
@@ -0,0 +1,243 @@
+package tracing
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/dop251/goja"
+	"go.k6.io/k6/js/common"
+	"go.k6.io/k6/js/modules"
+	httpmodule "go.k6.io/k6/js/modules/k6/http"
+	"go.k6.io/k6/metrics"
+)
+
+// Client represents a HTTP Client instrumenting the requests
+// it performs with tracing information.
+type Client struct {
+	vu modules.VU
+
+	// opts holds the client's configuration options.
+	opts options
+
+	// propagator holds the client's trace propagator, used
+	// to produce trace context headers for each supported
+	// formats: w3c, b3, jaeger.
+	propagator Propagator
+
+	// requestFunc holds the http module's request function
+	// used to emit HTTP requests in k6 script. The client
+	// uses it under the hood to emit the requests it
+	// instruments.
+	requestFunc HTTPRequestFunc
+}
+
+type (
+	// HTTPRequestFunc is a type alias representing the prototype of
+	// k6's http module's request function
+	HTTPRequestFunc func(method string, url goja.Value, args ...goja.Value) (*httpmodule.Response, error)
+)
+
+// NewClient instantiates a new tracing Client
+func NewClient(vu modules.VU, opts options) *Client {
+	rt := vu.Runtime()
+
+	// Instantiate the http module in our runtime and get its default exported object
+	httpModuleObj, ok := httpmodule.New().NewModuleInstance(vu).Exports().Default.(*goja.Object)
```

Comment:
Have you tried executing `require('k6/http')` instead. That will use the whole machinery with modules and will give you arguably a more stable code (as you won't depend on the internal API of k6, but of the external one that is used by k6 scripts).

The major downside is that at this time it means you will need to polyfill `require` for your tests, but I would expect it will work perfectly in the real case.

Reply:
Removed it and switched `require('k6/http').request` as recommended in another comment. 

I had troubles testing the module when it uses `require` under the hood, as `require` is only available from an initcontext, which, of course `goja.Runtime` doesn't have. That's why I resorted to integration tests. If you have an example of how to polyfill `require` available somewhere, that would be super valuable to me though 👍🏻 🙇🏻